### ISSUE TITLE
requires high performance gpu in macos

### ISSUE THREAD

**Issue by thisguychris** [gitterHQ/desktop#183]
Using the latest version 1.177 and I'm seeing it requires High performance GPU

Here's a screenshot of my activity monitor:
http://take.ms/yPOni

Possible fix? Set:
NSSupportsAutomaticGraphicsSwitching key to YES

**Comment by djspiewak** [comment]
Related to #179

**Comment by mydigitalself** [comment]
Slack's desktop app has the same thing btw.

@djspiewak how did you go about applying that setting?

**Comment by djspiewak** [comment]
@mydigitalself I haven't noticed any issues with Slack forcing the high performance GPU, and I've been keeping a *very* close eye on it.  As for how I applied the setting, I just opened the `Gitter.app/Contents/Info.plist` file and added the following XML to the end of the main `<dict>` block:

```xml
	<key>NSSupportsAutomaticGraphicsSwitching</key>
	<true/>
```

**Comment by jvasileff** [comment]
The `Info.plist` workaround works great for the GPU issue, but the tradeoff seems to be that logins are no longer persisted (I'm having to re-login every time gitter is launched.)

**Comment by firedev** [comment]
Indeed, that works, but why do I have to login again all the time? How is this related?

**Comment by djspiewak** [comment]
@firedev I'm assuming because it breaks the bundle checksum, and thus the signing, which means that Gatekeeper denies it access to Keychain.  Which is probably a good security feature, if that is indeed the case.  This modification really isn't that different from a piece of malware injecting code into an otherwise-trusted application and attempting to use it to escalate permissions.

**Comment by tscholak** [comment]
I noticed difference in behaviour between https://update.gitter.im/osx/Gitter-3.1.0.dmg and https://update.gitter.im/osx/latest (1.177). I can only speculate about the cause, since it is not clear to me what the difference is between these packages. Which one is preferred?

**Comment by MadLittleMods** [comment]
@tscholak v3.1.0 is preferred, NW.js backed, and is what the https://github.com/gitterHQ/desktop represents.

v1.177 is Safari-based, not open source. but is what we are currently pushing out on mac OS.

**Comment by asmeurer** [comment]
This also happens if you open gitter.im in Google Chrome. It would be great if it could be fixed for both instances.

**Comment by asmeurer** [comment]
By the way, https://gfx.io/ is a useful utility to telling you when an application is requiring the high performance (discrete) GPU.

**Comment by mydigitalself** [comment]
Doesn't work on new MBPs, sadly.

**Comment by djspiewak** [comment]
@asmeurer @mydigitalself I used Activity Monitor, which indicates whether or not the "High Performance" GPU is in use (and which applications require it) under the "Energy" tab.

**Comment by mydigitalself** [comment]
@djspiewak yeah i saw your screenshot, i couldn't find that column on my new new MBP + 10.12.4

**Comment by djspiewak** [comment]
@mydigitalself Here's the full window: http://i.imgur.com/Q94v5JE.png

**Comment by mydigitalself** [comment]
![screen shot 2017-04-11 at 23 07 33](https://cloud.githubusercontent.com/assets/1491887/24932933/b77e5e58-1f0b-11e7-93a6-fd230dbae71d.png)

**Comment by djspiewak** [comment]
I'm pretty sure your MacBook just doesn't have a discrete GPU.  So you're always using the integrated.

**Comment by asmeurer** [comment]
The newer MacBook Pros don't have this nonsense, just the older ones. If you go to "Apple -> About This Mac -> System report... -> Graphics/Displays" it will show you if you have two or just one graphics card.

**Comment by djspiewak** [comment]
Which nonsense? My MBP is 4 months old.

**Comment by mgol** [comment]
@MadLittleMods
> @tscholak v3.1.0 is preferred, NW.js backed, and is what the https://github.com/gitterHQ/desktop  represents.
> 
> v1.177 is Safari-based, not open source. but is what we are currently pushing out on mac OS.

What's the reason for not using the OSS version on macOS?

@asmeurer 

> The newer MacBook Pros don't have this nonsense, just the older ones.

My MacBook Pro is from 2015 and I definitely have this issue.

I've noticed that if I open an app that forces a dedicated graphics card, Gitter often starts blocking it as well. When the other app stops forcing the dedicated card, Gitter still holds onto it, never releasing it. After a restart it's fine again.

**Comment by MadLittleMods** [comment]
@mgol We want to improve the auto-update process before pushing out the NW.js version to macOS.

**Comment by mgol** [comment]
@MadLittleMods What are the current problems with auto-update? Is it totally unadvisable to download the 3.1.0 DMG and use it?

**Comment by MadLittleMods** [comment]
@mgol We want the experience to be like Chrome(downloads the update and on next startup the new version is in place). Currently, it bugs you about updating after we have downloaded the new update, then you have to watch it close the old version and pop up the new one (why we support multiple instances).

3.1.0 works great on macOS!

**Comment by mgol** [comment]
Thanks for the explanation! I'll try out the 3.1.0 on macOS then.

**Comment by riemers** [comment]
Downloaded 3.1.0 seems to resolved the issue for me. 👍

**Comment by jvasileff** [comment]
Yeah, 3.1.0 fixed the GPU issue for me too, but it introduces three new issues:

* There's a strange focus issue where the app doesn't respond to keyboard commands
* Sometimes the most recent post doesn't display without resizing the window
* The colors are off (overly vibrant); looks like a problem with color profiles
